src/components/renderer.cpp

```diff
@@ -160,14 +164,25 @@ renderer::renderer(
       m_log.info("Loaded font \"%s\" (name=%s, offset=%i, file=%s)", pattern, font->name(), offset, font->file());
       *m_context << move(font);
     }
+
+    m_log.trace("Activate root background manager");
+    m_background.activate(m_window, m_bar.outer_area(false));
   }
 
-  m_comp_bg = m_conf.get<cairo_operator_t>("settings", "compositing-background", m_comp_bg);
+  m_pseudo_transparency = m_conf.get<bool>("settings", "pseudo-transparency", m_pseudo_transparency);
+
+  // if we use pseudo-transparency, CAIRO_OPERATOR_SOURCE makes no sense as default
+  auto m_comp_bg_default = m_pseudo_transparency ? CAIRO_OPERATOR_OVER : m_comp_bg;
+  m_comp_bg = m_conf.get<cairo_operator_t>("settings", "compositing-background", m_comp_bg_default);
   m_comp_fg = m_conf.get<cairo_operator_t>("settings", "compositing-foreground", m_comp_fg);
   m_comp_ol = m_conf.get<cairo_operator_t>("settings", "compositing-overline", m_comp_ol);
   m_comp_ul = m_conf.get<cairo_operator_t>("settings", "compositing-underline", m_comp_ul);
   m_comp_border = m_conf.get<cairo_operator_t>("settings", "compositing-border", m_comp_border);
 
+  if (m_comp_bg == CAIRO_OPERATOR_SOURCE && m_pseudo_transparency) {
+    m_log.warn("pseudo-transparency may not work correctly with settings.compositing-background = source");
```

Comment:
This should check if something other than `over` was used. The warning should also tell them what to do, instead of what not to do.